inference-engine/src/transformations/include/transformations/op_conversions/gather_normalize_negative_indices.hpp

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <transformations_visibility.hpp>
+#include <ngraph/pass/graph_rewrite.hpp>
+
+namespace ngraph {
+namespace pass {
+
+    class TRANSFORMATIONS_API GatherNegativeIndicesNormalize;
+
+}  // namespace pass
+}  // namespace ngraph
+
+/**
+ * @ingroup ie_transformation_common_api
+ * @brief GatherNegativeIndicesNormalize check if indices value is negative and
+ * normalize it using ShapeOf->Add->Cast subgraph.
```

Comment:
Your transformation works only if the indices input is a scalar. Please, update the description of the transformation.

Reply:
BTW, why did you add such a restriction? Since the indices input is constant you actually can modify the constant of any shape, not just scalar... Just make sure that you modify negative values only and do not touch non-negative ones.

Reply:
Agree with @achetver. Additionally, we need a model to validate implementation a general case. Generation of synthetic model will take time. Probably it would be fine to insert TODO about how this transformation can be extended.

Reply:
Ok. Keep it as is. But I will take 10 points from Griffindor for this.